server/sonar-server/src/main/java/org/sonar/server/user/ws/SetHomepageAction.java

```diff
@@ -51,9 +70,55 @@ public void define(WebService.NewController controller) {
 
   @Override
   public void handle(Request request, Response response) throws Exception {
+    userSession.checkLoggedIn();
 
-    // WIP : Not implemented yet.
-    response.noContent();
+    final String type = request.param(PARAM_TYPE);
+    final String value = request.param(PARAM_VALUE);
+
+    if (PROJECT.toString().equals(type) && isBlank(value)) {
+      throw new IllegalArgumentException("type PROJECT requires a mandatory project key");
+    }
+
+    if (ORGANIZATION.toString().equals(type) && isBlank(value)) {
+      throw new IllegalArgumentException("type ORGANIZATION requires a mandatory project key");
+    }
+
+    String userLogin = userSession.getLogin();
+
+    String uuid = "";
+
+    try (DbSession dbSession = dbClient.openSession(false)) {
 
+      if (PROJECT.toString().equals(type)) {
+        com.google.common.base.Optional<ComponentDto> dto = dbClient.componentDao().selectByKey(dbSession, value);
+        if (dto.isPresent()) {
+          uuid = dto.get().uuid();
+        } else {
+          throw new IllegalStateException("No Project found for homepage key " + uuid);
+        }
+
+      }
+
+      if (ORGANIZATION.toString().equals(type)) {
+        Optional<OrganizationDto> dto = dbClient.organizationDao().selectByKey(dbSession, value);
+        if (dto.isPresent()) {
+          uuid = dto.get().getUuid();
+        } else {
+          throw new IllegalStateException("No Organization found for homepage key " + uuid);
```

Comment:
it must be an `IllegalArgumentException` because it's a user data provided

Reply:
done